lib/experimental/Navigation/Sidebar/Sidebar.tsx

```diff
@@ -14,9 +14,14 @@ export function Sidebar({ header, body, footer }: SidebarProps) {
   const { sidebarState, isSmallScreen } = useSidebar()
   const shouldReduceMotion = useReducedMotion()
   const [isScrolled, setIsScrolled] = useState(false)
+  const [hasScrollBottom, setHasScrollBottom] = useState(false)
 
   const handleScroll = (e: React.UIEvent<HTMLDivElement>) => {
-    setIsScrolled(e.currentTarget.scrollTop > 0)
+    const target = e.currentTarget
+    setIsScrolled(target.scrollTop > 0)
+    setHasScrollBottom(
+      target.scrollHeight - target.scrollTop - target.clientHeight > 1
```

Comment:
do you see this is the only way to proceed? Those methods are [known performance bottlenecks](https://gist.github.com/paulirish/5d52fb081b3570c81e3a/565c05680b27c9cfd9f5e971d295cd558c3e1843)